engine/common/timezones.py

```diff
@@ -0,0 +1,30 @@
+import pytz
+from rest_framework import serializers
+
+from common.api_helpers.exceptions import BadRequest
+
+
+def is_valid_timezone(timezone: str):
+    try:
+        return pytz.timezone(timezone)
+    except pytz.UnknownTimeZoneError:
+        return False
+
+
+def raise_exception_if_not_valid_timezone(timezone, Exception=BadRequest):
```

Comment:
nit: probably better to rename `Exception` to `exception_class` so it doesn't clash with the `Exception` class from builtins